src/Simplex/Messaging/Agent.hs

```diff
@@ -1265,6 +1290,25 @@ abortConnectionSwitch' c connId =
         _ -> throwError $ CMD PROHIBITED
       _ -> throwError $ CMD PROHIBITED
 
+synchronizeRatchet' :: AgentMonad m => AgentClient -> ConnId -> Bool -> m ConnectionStats
+synchronizeRatchet' c connId force = withConnLock c connId "synchronizeRatchet" $ do
+  withStore c (`getConn` connId) >>= \case
+    SomeConn _ (DuplexConnection cData@ConnData {ratchetSyncState} rqs sqs)
+      | ratchetSyncState `elem` ([RSAllowed, RSRequired] :: [RatchetSyncState]) || force -> do
+        -- check queues are not switching?
+        AgentConfig {e2eEncryptVRange} <- asks config
+        (pk1, pk2, e2eParams@(CR.E2ERatchetParams _ k1 k2)) <- liftIO . CR.generateE2EParams $ maxVersion e2eEncryptVRange
+        void $ enqueueRatchetKeyMsgs c cData sqs e2eParams
+        withStore' c $ \db -> do
+          setConnRatchetSync db connId RSStarted
+          deleteRatchet db connId
```

Comment:
I thought we wanted to keep it around in case it heals on some message? And create new when reply key arrives, unless it's healed by then?